pkg/util/template/engine.go

```diff
@@ -15,6 +15,11 @@ import (
 	"github.com/authgear/authgear-server/pkg/util/messageformat"
 )
 
+type StringWithLanguageTag struct {
```

Comment:
Name this RenderResult

```
type RenderResult struct {
  LanguageTag string
  String string
}
```

Note that in your original code, `languageTag` is a private type, so the outsider cannot see LanguageTag actually. You need to make it a String.